scss/_navbar.scss

```diff
@@ -139,13 +139,49 @@
 // Keep as a separate element so folks can easily override it with another icon
 // or image file as needed.
 .navbar-toggler-icon {
+  position: relative;
   display: inline-block;
-  width: 1.5em;
-  height: 1.5em;
-  vertical-align: middle;
-  background-repeat: no-repeat;
-  background-position: center;
-  background-size: 100%;
+  width: $navbar-toggler-font-size;
+  height: $border-width * 2;
+  vertical-align: super;
+  background: currentColor;
+  @include transition(background .3s ease);
+
+  &::before,
+  &::after {
+    position: absolute;
+    left: 0;
+    width: inherit;
+    height: inherit;
+    content: "";
+    background: currentColor;
+    transform: rotate3d(0, 0, 1, 0);
+    transform-origin: 50% 50%;
+    @include transition(.3s ease);
```

Comment:
Specifying `transform` and `top` as transitionned properties might be better, wouldn't it?

Reply:
`top` isn't performant to transition IIRC. `transform` is though.

Reply:
So that'd need a complex `transform` with both `rotate3d()` and `translate3d()` to drop the `top` part.

Reply:
Let's wait for @mdo and I'm pretty sure we'll figure something out altogether.

I still think that this should be toggleable somehow and perhaps respect the reduced motion?

Reply:
This is handled since it's using our `transition()` mixin.

Reply:
OK, so this leaves the option to toggle the animation or not. The class approach would mean it would be toggleable without recompiling, but I don't think we do this anywhere else (yet).